### PR TITLE
docs(atom-with-broadcast): update example code to work with TypeScript

### PR DESCRIPTION
**Pull request by hoangvu12**
## Related Bug Reports or Discussions

None

## Summary

- The old code used the library's old TypeScript API. This update fixes it.

## Check List

- [x] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

File: docs/recipes/atom-with-broadcast.mdx [hunk 1]
```diff
@@ -13,17 +13,22 @@ By using the BroadcastChannel API, you can enable basic communication between br
...
   channel.onmessage = (event) => {
     listeners.forEach((l) => l(event))
   }
 
-  const broadcastAtom = atom<Value, { isEvent: boolean; value: Value }>(
+  const broadcastAtom = atom<
+    Value,
+    [{ isEvent: boolean; value: SetStateAction<Value> }],
+    void
+  >(
```
Comment (dai-shi) [inline docs/recipes/atom-with-broadcast.mdx:31]: Can we remove type here and add the type for `update` below? That's the recommended way.
```suggestion
  const broadcastAtom = atom(
```

File: docs/recipes/atom-with-broadcast.mdx [hunk 2]
```diff
@@ -33,23 +38,29 @@ export function atomWithBroadcast<Value>(key: string, initialValue: Value) {
...
     return () => {
       listeners.delete(listener)
     }
   }
-  const returnedAtom = atom<Value, Value>(
+
+  const returnedAtom = atom<Value, [SetStateAction<Value>], void>(
     (get) => get(broadcastAtom),
-    (get, set, update) => {
+    (_, set, update) => {
```
Comment (dai-shi) [inline docs/recipes/atom-with-broadcast.mdx:56]: We often do this:
```suggestion
    (_get, set, update) => {
```

File: docs/recipes/atom-with-broadcast.mdx (hunk 1 above)
Comment (hoangvu12) [reply docs/recipes/atom-with-broadcast.mdx:31]: Sorry, what do you mean by `add the type for "update" below`? Do you mean doing this?

```ts
const broadcastAtom = atom(
  (get) => get(baseAtom),
  (get, set, update: { isEvent: boolean; value: SetStateAction<Value> }) => {
    set(baseAtom, update.value)

    if (!update.isEvent) {
      channel.postMessage(get(baseAtom))
    }
  }
)
```

Comment (dai-shi) [reply docs/recipes/atom-with-broadcast.mdx:31]: Yes, exactly! Sorry for the confusion.

**Review by dai-shi — APPROVED** [review]
LGTM
Thanks for your contribution!

If you find anything else in docs to improve, feel free to send PRs.

**Pull request merged**